components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java

```diff
@@ -934,6 +935,12 @@ private void callDefaultProvisioningHandler(String username, AuthenticationConte
             excludeUnmappedRoles = Boolean
                     .parseBoolean(IdentityUtil.getProperty(SEND_ONLY_LOCALLY_MAPPED_ROLES_OF_IDP));
         }
+
+        IdentityUtil.threadLocalProperties.get()
+                .put(IdentityEventConstants.EventProperty.STEP_ID, context.getCurrentStep());
+        IdentityUtil.threadLocalProperties.get()
+                .put(IdentityEventConstants.EventProperty.CURRENT_AUTHENTICATOR, context.getCurrentAuthenticator());
+
```

Comment:
Remove these within the same class in a finally block. Else it's very hard to troubleshoot and maintain.
Also better move to a util indicating that this populate thread local context with additional info for events

Reply:
Fixed 8d2f7ed73ccf54d1cbc7c85e4a28a8b696e2169f